src/dataflow-types/src/client/replicated.rs

```diff
@@ -251,6 +273,11 @@ where
     }
 
     async fn recv(&mut self) -> Result<Option<ComputeResponse<T>>, anyhow::Error> {
+        // If we have a pending response, we should send it immediately.
```

Comment:
I guess this works fine because we're single threaded, can we add a comment as to why this works as expected? Specifically, why we can't be sitting in one of the awaits below.